Sources/XMLCoder/Box/Box.swift

```diff
@@ -0,0 +1,238 @@
+        case .unsignedInteger(_): return true
+        case .floatingPoint(_): return true
+        case .string(_): return true
+        case .array(_): return false
+        case .dictionary(_): return false
+        }
+    }
+    
+    var bool: BoolBox? {
+        guard case .bool(let box) = self else {
+            return nil
+        }
+        return box
+    }
+    
+    var decimal: DecimalBox? {
+        guard case .decimal(let box) = self else {
+            return nil
+        }
+        return box
+    }
+    
+    var signedInteger: SignedIntegerBox? {
+        guard case .signedInteger(let box) = self else {
+            return nil
+        }
+        return box
+    }
+    
+    var unsignedInteger: UnsignedIntegerBox? {
+        guard case .unsignedInteger(let box) = self else {
+            return nil
+        }
+        return box
+    }
+    
+    var floatingPoint: FloatingPointBox? {
+        guard case .floatingPoint(let box) = self else {
+            return nil
+        }
+        return box
+    }
+    
+    var string: StringBox? {
+        guard case .string(let box) = self else {
+            return nil
+        }
+        return box
+    }
+    
+    var array: ArrayBox? {
+        guard case .array(let box) = self else {
+            return nil
+        }
+        return box
+    }
+    
+    var dictionary: DictionaryBox? {
+        guard case .dictionary(let box) = self else {
+            return nil
+        }
+        return box
+    }
+    
+    var xmlString: String {
+        switch self {
+        case .null(let box): return box.xmlString
+        case .bool(let box): return box.xmlString
+        case .decimal(let box): return box.xmlString
+        case .signedInteger(let box): return box.xmlString
+        case .unsignedInteger(let box): return box.xmlString
+        case .floatingPoint(let box): return box.xmlString
+        case .string(let box): return box.xmlString
+        case .array(let box): return box.xmlString
+        case .dictionary(let box): return box.xmlString
+        }
+    }
+    
+    init() {
+        self = .null(.init())
+    }
+    
+    init(_ unboxed: Bool) {
+        self = .bool(.init(unboxed))
+    }
+    
+    init(_ unboxed: Decimal) {
+        self = .decimal(.init(unboxed))
+    }
+    
+    init<Integer: SignedInteger>(_ unboxed: Integer) {
+        self = .signedInteger(.init(unboxed))
+    }
+    
+    init<Integer: UnsignedInteger>(_ unboxed: Integer) {
+        self = .unsignedInteger(.init(unboxed))
+    }
+    
+    init<Float: BinaryFloatingPoint>(_ unboxed: Float) {
+        self = .floatingPoint(.init(unboxed))
+    }
+    
+    init(_ unboxed: String) {
+        self = .string(.init(unboxed))
+    }
+    
+    init(_ unboxed: [Box]) {
+        self = .array(.init(unboxed))
+    }
+    
+    init(_ unboxed: [String: Box]) {
+        self = .dictionary(.init(unboxed))
+    }
+    
+    internal func unbox() throws -> Bool? {
```

Comment:
probably could remove `internal` in this function and below?

Reply:
Fixed.